src/pages/TradePage/TradePage.tsx

```diff
@@ -205,6 +212,44 @@ export const TradeChart = ({
     assetIds,
   ]);
 
+  useEffect(() => {
+    const lastRecordOutdatedBy = 60000;
+
+    if (
+      !isVisible ||
+      historicalBalancesLoading ||
+      datasetRefreshing
+    )
+      return;
+
+    const refetchHistoricalBalancesData = () => {
+      if (
+        !datasetRefreshing &&
```

Comment:
this would also prevent the from/to from being updated, while the query is already in flight. I don't think this is 100% desired behavior, is it?

Reply:
also please use `networkStatus` here